logger/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+use cstr::*;
+use log::LevelFilter;
+use log4rs::encode::pattern::PatternEncoder;
+use log4rs::Handle;
+use log4rs::{
+    append::rolling_file::{
+        policy::compound::{
+            roll::fixed_window::FixedWindowRoller, trigger::size::SizeTrigger, CompoundPolicy,
+        },
+        RollingFileAppender,
+    },
+    config::{Appender, Config, Root},
+};
+use shared_sql_utils::odbcinst::SQLGetPrivateProfileString;
+use std::path::{Path, PathBuf};
+
+const LOG_FILE_SIZE: u64 = 1024 * 500;
+
+#[derive(Debug)]
+pub struct Logger {
+    handle: Handle,
+    log_dir: PathBuf,
+}
+
+impl Logger {
+    pub fn new() -> Option<Self> {
+        // Due to numerous reasons why the logger could fail to initialize, we wrap it in a catch_unwind
+        // so that logger failure does not cause our dll to crash.
+        match std::panic::catch_unwind(|| {
+            let mut buffer = [0u8; 1024];
+            // TODO: use cargo env to get version info
+            unsafe {
+                SQLGetPrivateProfileString(
+                    to_char_ptr("MongoDB Atlas SQL ODBC Driver").0,
+                    to_char_ptr("Driver").0,
+                    to_char_ptr("").0,
+                    buffer.as_mut_ptr(),
+                    buffer.len() as i32,
+                    to_char_ptr("odbcinst.ini").0,
+                )
+            };
```

Comment:
Why SQLGetPrivateProfileString instead of SQLGetPrivateProfileStringW? Just a choice, but easily converted if we want.